zcash_client_sqlite/src/wallet.rs

```diff
@@ -765,21 +810,24 @@ pub(crate) fn insert_block(
     block_height: BlockHeight,
     block_hash: BlockHash,
     block_time: u32,
-    commitment_tree: &CommitmentTree,
+    sapling_commitment_tree_size: Option<u64>,
 ) -> Result<(), SqliteClientError> {
-    let mut encoded_tree = Vec::new();
-    write_commitment_tree(commitment_tree, &mut encoded_tree).unwrap();
-
     let mut stmt_insert_block = conn.prepare_cached(
-        "INSERT INTO blocks (height, hash, time, sapling_tree)
-            VALUES (?, ?, ?, ?)",
+        "INSERT INTO blocks (
+            height,
+            hash,
+            time,
+            sapling_commitment_tree_size,
+            sapling_tree
+        )
+        VALUES (?, ?, ?, ?, x'00')",
```

Comment:
If the dummy value parses as a `CommitmentTree` that is _worse_ because it could lead a user to believe they've parsed a valid commitment tree when they've actually parsed the "tree absent" indicator. Instead, using a value that cannot parse as a valid `CommitmentTree` is the correct thing to do here, because then parsing will return an error (and not panic.) Since after this change there should be no code left that parses the `sapling_tree` value, this serves as a good check against incorrect behavior.